src/GraphQL/Execution/DocumentExecuter.cs

```diff
@@ -633,7 +637,7 @@ public bool DoesFragmentConditionMatch(ExecutionContext context, IHaveFragmentTy
                 return false;
             }
 
-            if (conditionalType.Equals(type))
+            if (conditionalType.ToString() == type.ToString())
```

Comment:
I'm fairly sure we want to use `Equals` vs. a `ToString` here.  Do you have a use case?  Unit tests to prove?